### PR TITLE
Fixed #28781 -- Added QuerySet.values()/values_list() support for union(), difference(), and intersection().

### PR DESCRIPTION
**Pull request by felixxm**
https://code.djangoproject.com/ticket/28781

### CODE REVIEW

File: django/db/models/sql/compiler.py [hunk 1]
```diff
@@ -407,6 +407,10 @@ def get_combinator_sql(self, combinator, all):
         parts = ()
         for compiler in compilers:
             try:
+                # Set the selects defined on the combined query by a call to
```
Comment (timgraham) [inline django/db/models/sql/compiler.py]: Maybe you could try rephrasing this comment. I'm having trouble understanding it and suggesting an alternative.

File: docs/releases/1.11.8.txt [hunk 2]
```diff
@@ -11,3 +11,6 @@ Bugfixes
 
 * Reallowed, following a regression in Django 1.10, ``AuthenticationForm`` to
   raise the inactive user error when using ``ModelBackend`` (:ticket:`28645`).
+
+* Fixed ``QuerySet.values()/values_list()`` for ``union()``, ``difference()``,
```
Comment (timgraham) [inline docs/releases/1.11.8.txt]:
```
Added support for ``QuerySet.values()`` and ``values_list()`` ...
```

**Comment by felixxm** [comment]
@timgraham Thanks for review. I updated docs and comments.

File: docs/ref/models/querysets.txt [hunk 3]
```diff
@@ -813,10 +813,16 @@ duplicate values, use the ``all=True`` argument.
 of the type of the first ``QuerySet`` even if the arguments are ``QuerySet``\s
 of other models. Passing different models works as long as the ``SELECT`` list
 is the same in all ``QuerySet``\s (at least the types, the names don't matter
-as long as the types in the same order).
+as long as the types in the same order). In such cases, you must use column
+names from the first ``QuerySet`` in ``order_by()``. For example::
```
Comment (timgraham) [inline docs/ref/models/querysets.txt]: "in QuerySet methods applied to the resulting QuerySet"?
(besides order_by(), it also applies to values/values_list)

File: docs/ref/models/querysets.txt [hunk 4]
```diff
@@ -813,10 +813,16 @@ duplicate values, use the ``all=True`` argument.
 of the type of the first ``QuerySet`` even if the arguments are ``QuerySet``\s
 of other models. Passing different models works as long as the ``SELECT`` list
 is the same in all ``QuerySet``\s (at least the types, the names don't matter
-as long as the types in the same order).
+as long as the types in the same order). In such cases, you must use column
```
Comment (timgraham) [inline docs/ref/models/querysets.txt]: the column names

File: django/db/models/sql/compiler.py [hunk 5]
```diff
@@ -407,6 +407,11 @@ def get_combinator_sql(self, combinator, all):
         parts = ()
         for compiler in compilers:
             try:
+                # If columns list is limited, then all combined queries must
```
Comment (timgraham) [inline django/db/models/sql/compiler.py]: If the columns list is limited, ...

**Comment by felixxm** [comment]
@timgraham Thanks again!

**Pull request merged**